### PR TITLE
LPD-34390 Internal Server Error when retrieving a structured content with localizable fields in a field group

### PR DESCRIPTION
**Pull request by jkappler**
Internal Server Error when retrieving a structured content with localizable fields in a field group

# What is this trying to solve?

[Link to ticket](https://issues.liferay.com/browse/LPD-34390)

Avoid throwing a NPE when the value of a field is null

# How am I fixing it?

Control the NPE

# How can you verify that it works?

Follow the steps in the ticket or run the provided tests

### CODE REVIEW

File: modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/ContentFieldUtil.java [hunk 1]
```diff
@@ -91,10 +91,14 @@ public static ContentField toContentField(
 							return null;
 						}
 
-						Map<String, ContentFieldValue> map = new HashMap<>();
-
 						Value value = ddmFormFieldValue.getValue();
 
+						if (value == null) {
+							return new HashMap<>();
+						}
```
Comment (AliciaGarciaGarcia) [inline modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/ContentFieldUtil.java:96]: suggestion: 

On line 77 and 91 it's returning a null. maybe returning a null is more standardised on this.

then you can put on the 99 line if unified

**Comment by jkappler** [comment]
@AliciaGarciaGarcia, added the change that we talked about, thanks! ❤️

File: modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/ContentFieldUtil.java (hunk 1 above)
Comment (jkappler) [reply modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/ContentFieldUtil.java:96]: ❤️

Comment (jkappler) [reply modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/ContentFieldUtil.java:96]: Changes made in the past :P

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:report:67741680

**Comment by jkappler** [comment]
ci:forward:force

**Comment by jkappler** [comment]
@AliciaGarciaGarcia, failures are not related

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by jkappler** [comment]
ci:forward:force

**Pull request closed without merging**